### PR TITLE
Improve debugging error messages

### PR DESCRIPTION
**Pull request by rwest**
### Motivation or Problem
As described in #2030, the error messages and strack traces are not always super helpful in finding and fixing the cause of the problem. This PR attempts to add a few more diagnostics.

### Description of Changes
The main approach is to catch an exception as it passes through a section of code that knows something helpful, log a diagnostic error message, and then re-raise the exception so that it carries on as before. 
For example, if you get a bad atom type when trying to estimate adsorption thermo for a species that was included in a reaction library, then it's helpful to know which species, which reaction library, as well as which atom.  But the atom typing code doesn't know this info when it raises the exception, so we have to catch and re-raise it a few times. 

### Testing
I ran the unit tests. Then added a unit test that causes several of the errors that trigger the new debugging messages. All tests now pass.

### Reviewer Tips
As this code is only supposed to work when something goes wrong, testing it is interesting.
I hope that @tignchenlee can try this branch on her broken reaction libraries, input files, etc. (before she figured out how to fix them) and see if the new error messages work and would have helped. Perhaps @katrinblondal and @mazeau have some input files with recent versions of databases etc. (or before the bidentate problems were fixed) that cause exceptions too.
_Update: I added unit tests._

### Copy this PR! (or add commits)
If you have recently come across a bug (or found one in the issue tracker) feel free to take this approach (catch an exception, log something helpful, and raise it again) and apply it elsewhere. This could be just the start. The focus was on thermo problems.

### CODE REVIEW

**Review by mazeau — APPROVED** [review]
This looks great. Is there a reason for switching the `.format` to `f` strings?

**Comment by rwest** [comment]
> Is there a reason for switching the `.format` to `f` strings?

f-strings are easier to read, easier to debug, and faster to execute. Generally better.
https://realpython.com/python-f-strings/

File: rmgpy/rmg/model.py
Comment (rwest) [inline rmgpy/rmg/model.py:425]:
```suggestion
            logging.error(f"Error when making species in reaction {forward:s} from {forward.family:s}")
```

Shall I pretend I left this mistake in to check our code review process?

Comment (mazeau) [reply rmgpy/rmg/model.py]: Ah, I should have caught this! Sorry

**Review by mazeau — APPROVED** [review]
Ok, I see no issues with merging this in

**Comment by rwest** [comment]
Yes! I got the code coverage test to pass (even though this is updating error-handling code that therefore shouldn't normally be run). OK, now all tests pass, this could be reviewed and merged?  If folks want to add more debugging messages of this type, then can be in a new PR.

**Review by mjohnson541 — DISMISSED** [review]
This looks good to me, once it passes RMG-tests should be good to merge.

**Comment by rwest** [comment]
The previous commit passed all tests. This one has stalled for some reason, and I can't see how to trigger it again.

**Comment by rwest** [comment]
@mjohnson541 said good to merge once tests pass. The only change since then was a pretend change (tweak a commit message) to trigger the build scripts again, which had stalled. Now they have completed. All passed, so I merged.

**Pull request merged**
